pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -351,11 +353,12 @@ func (d *defaultDriver) Reconcile(ctx context.Context) *reconciler.Results {
 		return results.WithError(err)
 	}
 
-	// reconcile an empty File based settings Secret if it doesn't exist
 	if d.Version.GTE(filesettings.FileBasedSettingsMinPreVersion) {
-		err = filesettings.ReconcileEmptyFileSettingsSecret(ctx, d.Client, d.ES, true)
+		requeue, err := maybeReconcileEmptyFileSettingsSecret(ctx, d.Client, d.LicenseChecker, &d.ES, d.OperatorParameters.OperatorNamespace)
 		if err != nil {
 			return results.WithError(err)
+		} else if requeue {
+			results.WithReconciliationState(defaultRequeue.WithReason("File settings secret is not reconciled"))
```

Comment:
```suggestion
			results.WithReconciliationState(
				defaultRequeue.WithReason(
					fmt.Sprintf("This cluster is targeted by at least one StackConfigPolicy, expecting Secret %s to be created by StackConfigPolicy controller", esv1.FileSettingsSecretName(d.ES.Name))),
			)
```

Reply:
changed in 6fd2b1135c88d031eb9e8181ee474eec04b7cee1